### ISSUE TITLE
Agena flight pack balancing

### ISSUE THREAD

**Issue by jwvanderbeck** [KSP-RO/RealismOverhaul#1105]
So there are two flight packs defined in the FASA configs.  One for an Agena-B flightpack and one for the Agena-D.

Now I have had trouble finding any solid sources, but I did peruse the link to the -D document indicated in the configs.

Right now the -D config in RO has both the "aft rack" which stores nitrogen and RCS, as well as a pair of SPS units.  The -B config uses the same model which visually has the two SPS units, but doesn't actually include the SPS units in the config.

The -D masses 0.206, while the -B masses 0.18891

According to the inline references for mass on the -D that mass of 0.206 is a mass of 0.091 plus 0.115 for the two SPS.  So it seems to me that if the -B model is NOT going to have the SPS then it should mass around the 0.091 number.  Either that or it should get the SPS.

Right now the -B is essentially a 200kg wet nitrogen tank with RCS.  You can do the exact same thing with a proc tank and attitude jets, with the exact same configuration and thrust, for 35kg.

**Comment by jwvanderbeck** [comment]
Is this something that RF can config to allow swapping out the various SPS units as desired, and adjusting the mass accordingly?  If so, I would give the -B model the option of no SPS or 16lb SPS, and then add the 200lb SPS to the -D as an option.

Then the player can choose just like they could in real life.

**Comment by jwvanderbeck** [comment]
Ping?

**Comment by stratochief66** [comment]
You say that the 2 parts already have different masses (0.206T & 0.18891T) so I don't see any reason they couldn't be changed.

It seems reasonable to me to set the Agena-B to be lighter, as it doesn't have the SPS. Without the SPS and with just nitrogen it is still capable of pointing the Agena-B & Payload before Agena re-ignition.

I'm not sure if Rokker has a Github handle, but he sees Agena's fairly regularly and might have some information on whether the Agena B should have an SPS of its own or not.

**Comment by jwvanderbeck** [comment]
Err my point is the masses are not that different and the -B imho is very heavy for lacking the SPS.  

If the -D is massed, according to the comments as "0.091 t + 0.115 t for two SPS units (1)" then why is the -B *without* the SPS not 0.091.

Right now it is stupid to ever use the -B flightpack because it is 200kg wet and you can build the exact same thing using proc tanks and attitude thrusters, for about 35kg.  And the thrusters have the same control and the exact same thrust levels.  For 200kg if you built it yourself you would have tons more prop than the flightpack carries.

**Comment by raidernick** [comment]
It doesn't matter what the flightpack masses to. Procedural tanks of the same size will always weigh different than the real life counterparts. It only matters that the entire craft weighs what it does IRL and it does. It's your prerogative to whether or not you want to make your own custom lighter parts but then the spacecraft is not realistic anymore.

**Comment by stratochief66** [comment]
@raidernick do you have a good lead on the masses and capabilities of the various Agena variants? I believe part of the problem here is that we don't have any good sources for the real thing.

Certainly you are correct that there will always be differences between the proc version of something and the historically sourced recreation.

**Comment by jwvanderbeck** [comment]
Fair enough, but then the comments in the CFG still don't make any sense to me.

How can the -D be "0.091 base, and 0.115 for the SPS" while the -B without SPS is 0.18891 without the SPS.  If the -D is saying it is 0.091 without the SPS why is the -B not 0.091?

That is the part that is confusing me.  In otherwords if the SPS masses 0.115 as the comment states, and the -D masses 0.206, then at worst the -B should be around (0.206 - 0.115) or 0.091.

Also you say that it only matters that the *entire* craft masses correct which is fair but how do you know the proportions are correct? Just curious not accusing anyone here.

**Comment by jwvanderbeck** [comment]
Do we have sources that say for a certainty that the -B did not have an SPS option?  Because it seems to me that if the -B did have SPS then the masses would make a lot more sense.

**Comment by raidernick** [comment]
The agena in the FASA pack is the agena D, it looks nothing like the agena B. The agena b has no flight pack or docking port. We are making the best with the parts in the pack. http://historicspacecraft.com/Photos/Upper_Stage/Agena-B_UH_RK_2008_3.jpg
The agena in the FASA pack requires the flight pack to work correctly as the meshes were not designed to fit together properly without it, so this entire argument is moot. I am working on making other agena variants but until then this is the best you've got.

**Comment by jwvanderbeck** [comment]
Actually we've been discussing it in IRC and found sources that indicate
the SPS was available on the Agena B for at least some DOD missions.

- John Vanderbeck
- http://www.johnvanderbeck.com

**Comment by raidernick** [comment]
I would also like to point out I did not mass any of the parts like that. I did the original configs and "somebody" decided they knew better and went and changed all the part masses despite not having any sources, so you should check the changelog for it and ask them why they changed it.

**Comment by jwvanderbeck** [comment]
source: http://claudelafleur.qc.ca/Spacecrafts-1962.html

> Samos 9 / PVP 853
> Spacecraft: 	Program 201 (BJ) ; FTV 2403 / AFP-201 PVP 853
> Chronologies:	1962 payload #42 ; 1962-030A ; 201st spacecraft.
> Type:	Reconnaissance
> Sponsor:	U.S. National Reconnaissance Office
> Launch:	18 July 1962 at 0h51 UTC, from Vandenberg Air Force Base's LC-1-1, by an Atlas-Agena B (Atlas LV-3A 120D / Agena B FTV 2403).
> Orbit:	
> Reentered:	27 July 1962
> Mission:	FTV 2403 (Program 698BJ Vehicle 3, with payload PVP 853) was launched from Point Arguello. The Agena secondary 
> propulsion system (SPS) failed to operate due to an electrical short. The planned orbit was 213 x 256 km, and FTV 2403 achieved 217 x 227 km. The satellite operated for at least 18 orbits.  
>      The retrofire appears to have failed, but the SRV separated anyway and was left in a similar 184 x 234 km orbit. The Agena reentered after 7 days, on July 25, and the SRV reentered on July 27 The spacecraft carried a neutron albedo experiment in addition to the USAF intelligence payload. 
> Capsule failed to deorbit.

IMHO the best idea is to make the SPS an engine config with the following options:
Agena-B: No SPS (mass adjusted), 16lb thrust SPS
Agena-D: Non SPS (mass adjusted), 16lb thrust SPS, 200lb thrust SPS

**Comment by stratochief66** [comment]
Looks reasonable @jwvanderbeck. Now turn it into a PR :P
